src/rdf.rs

```diff
@@ -0,0 +1,138 @@
+use log::{info, error};
+
+use oxigraph::io::{GraphFormat};
+use oxigraph::store::{Store, QuadIter, SerializerError};
+use oxigraph::model::*;
+use oxigraph::model::vocab::{rdf, xsd};
+
+use crate::vocab::{dcterms, dcat, dqv, prov};
+
+// Parse Turtle RDF and load into store
+pub fn parse_turtle(turtle: String) -> Store {
+    info!("Loading turtle graph");
+
+    let store = Store::new().unwrap();
+    match store.load_graph(turtle.as_ref(), GraphFormat::Turtle, GraphNameRef::DefaultGraph, None) {
+        Ok(_) => info!("Graph loaded successfully"),
+        Err(e) => error!("Loading graph failed {}", e)
+    }
+
+    return store;
+}
+
+// Retrieve datasets
+pub fn list_datasets(store: &Store) -> QuadIter {
+    store.quads_for_pattern(None, Some(rdf::TYPE), Some((dcat::DATASET_CLASS).into()), None)
+}
+
+// Retrieve distributions of a dataset
+pub fn list_distributions(dataset: NamedNodeRef, store: &Store) -> QuadIter {
+    store.quads_for_pattern(Some((dataset).into()), Some((dcat::DISTRIBUTION).into()), None, None)
+}
+
+// Retrieve distribution formats
+pub fn list_formats(distribution: NamedOrBlankNodeRef, store: &Store) -> QuadIter {
+    store.quads_for_pattern(Some((distribution).into()), Some((dcterms::FORMAT).into()), None, None)
+}
+
+// Retrieve distribution media-types
+pub fn list_media_types(distribution: NamedOrBlankNodeRef, store: &Store) -> QuadIter {
+    store.quads_for_pattern(Some((distribution).into()), Some((dcat::MEDIA_TYPE).into()), None, None)
+}
+
+// Retrieve dataset namednode
+pub fn get_dataset_node(store: &Store) -> Option<NamedNode> {
+    match list_datasets(&store).next() {
+        Some(d) => {
+            match d {
+                Ok(d) => {
+                    match d.subject {
+                        Subject::NamedNode(n) => Some(n),
+                        _ => None                      
+                    }
+                },
+                Err(_) => None
+            }            
+        },
+        None => None       
+    }
```

Comment:
Kan vurdere litt _magi_ her, for å forenkle litt :smile: 
```suggestion
    list_datasets(&store).next().and_then(|d| match d {
        Ok(Quad {
            subject: Subject::NamedNode(n),
            ..
        }) => Some(n),
        _ => None,
    })
```